apps/starter-nextjs-trpc-clerk/pages/index.tsx

```diff
@@ -0,0 +1,134 @@
+import Head from "next/head";
+import { SignedIn, SignedOut, SignOutButton, useAuth } from "@clerk/nextjs";
+
+import { RouterOutputs, trpc } from "../utils/trpc";
+import { AddTodo } from "../components/AddTodo/AddTodo";
+import { TodoItem } from "../components/TodoItem/TodoItem";
+import { Card } from "../components/Card/Card";
+import { Welcome } from "../components/Welcome/Welcome";
+
+export default function Home() {
+  const utils = trpc.useContext();
+  const { isSignedIn } = useAuth();
+
+  const { data: todos } = trpc.todo.list.useQuery(undefined, {
+    enabled: isSignedIn,
+  });
+
+  const { mutate: addTodo } = trpc.todo.add.useMutation({
+    onMutate: async (newTodo) => {
+      await utils.todo.list.cancel();
+
+      const previousTodos = await utils.todo.list.getData();
```

Comment:
according to my TS Server when from the little I know about tRPC, I think this method is sync.

```suggestion
      const previousTodos = utils.todo.list.getData();
```